emcc.py

```diff
@@ -2394,7 +2386,7 @@ def generate_html(target, options, js_target, target_basename,
 %s
           };
           emterpretXHR.send(null);
-''' % (shared.Settings.EMTERPRETIFY_FILE, script.inline)
+''' % (shared.JS.get_subresource_location(shared.Settings.EMTERPRETIFY_FILE), script.inline)
```

Comment:
Cool, sounds good; I just pushed another commit to handle all the parsing in the HTML output and update the documentation. The only remaining CSP quirks are:

* Embedding asm.js requires `script-src 'unsafe-inline'` (out of `data:`, `'unsafe-eval'`, and `'unsafe-inline'`, that one seemed like the least unusual to require)

* Using Workers requires `child-src blob:`

As far as mem init 0, yep, there's no risk of CSP breaking that now that it's parsed without XHR in all cases.